keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,405 @@
+
+  // The map key is the input resource name.
+  Strategy map[corev1.ResourceName]ResourceTransformationStrategy `json:"strategy"`
+}
+```
+
+Grouping by output makes it easier for the cluster admin to see/specify
+all the resources that map into a given output as a unit.  However it
+comes at the cost of a separate map to specify strategy.
+It may be confusing that the `Transformations` map is keyed by
+output name as the primary key while `Strategy` is keyed by input name.
+
+### Comparison
+
+Consider the following list of 4-tuples
+
+* (`nvidia.com/mig-1g.5gb`, `kueue.x-k8s.io/accelerator-memory`, `5G`, `replace`)
+* (`nvidia.com/mig-2g.10gb`, `kueue.x-k8s.io/accelerator-memory`, `10G`, `replace`)
+* (`nvidia.com/mig-4g.20gb`, `kueue.x-k8s.io/accelerator-memory`, `20G`, `replace`)
+* (`nvidia.com/mig-1g.5gb`, `kueue.x-k8s.io/credits`, `10`, `replace`)
+* (`nvidia.com/mig-2g.10gb`, `kueue.x-k8s.io/credits`, `15`, `replace`)
+* (`nvidia.com/mig-4g.20gb`, `kueue.x-k8s.io/credits`, `30`, `replace`)
+* (`cpu`, `kueue.x-k8s.io/credits`, `1`, `retain`)
+
+Option 1
+```yaml
+resourceTransformationMatrix:
+  transformations:
+    nvidia.com/mig-1g.5gb:
+      strategy: Replace
+      conversionBaseValues:
+        kueue.x-k8s.io/accelerator-memory: 5G
+        kueue.x-k8s.io/credits: 10
+    nvidia.com/mig-2g.10gb:
+      strategy: Replace
+      conversionBaseValues:
+        kueue.x-k8s.io/accelerator-memory: 10G
+        kueue.x-k8s.io/credits: 15
+    nvidia.com/mig-4g.20gb:
+      strategy: Replace
+      conversionBaseValues:
+        kueue.x-k8s.io/accelerator-memory: 20G
+        kueue.x-k8s.io/credits: 30
+    cpu:
+      strategy: Retain
+      conversionBaseValues:
+        kueue.x-k8s.io/credits: 1
+```
+
+Option 2
+```yaml
+resourceTransformationMatrix:
+  transformations:
+    kueue.x-k8s.io/accelerator-memory:
+      nvidia.com/mig-1g.5gb: 5G
+      nvidia.com/mig-2g.10gb: 10G
+      nvidia.com/mig-4g.20gb: 20G
+    kueue.x-k8s.io/credits:
+      nvidia.com/mig-1g.5gb: 10
+      nvidia.com/mig-2g.10gb: 15
+      nvidia.com/mig-4g.20gb: 30
+      cpu: 1
+  strategy:
+    nvidia.com/mig-1g.5gb: Replace
+    nvidia.com/mig-2g.10gb: Replace
+    nvidia.com/mig-4g.20gb: Replace
+    cpu: Retain
+```
+
+Option 2 is likely to be terser, but I find Option 1 slightly more intutive because
+I tend to think of functions in the "forward direction" (mapping inputs to outputs).
+Either is viable and has similar implementation complexity.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+None
+
+#### Unit Tests
+
+Unit tests of the helper function invoked from `workload.AdjustResources` will be added.
+
+Unit tests of loading the `ResourceTransformationMatrix` from a `Configuration` will be added.
+
+#### Integration tests
```

Comment:
How will Kueue handle an update to the transformation matrix followed by a controller restart? What will happen to existing Workloads?

Reply:
Existing workloads would be subject to the new matrix when `AdjustResources` is invoked on them (or more precisely a deepCopy of them).  This would mainly happen during the processing of an Update event (workload controller) or in the corner case of an existing workload whose local queue was created after it (queue.AddLocalQueue).

This looks plausible to me, but I do not claim to fully understand this part of the system and may be missing something.